File: Hash Tables/first_recurring_char.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

int find_the_recurring_character(const std::vector<int>& nums) {
    std::unordered_map<int, bool> seen;

    for (int num : nums) {
        if (seen.find(num) != seen.end()) {
            return num;
        }
        seen[num] = true;
    }

    return -1; 
}

int main() {
    std::vector<int> array = {2, 1, 1, 2, 3, 5, 1, 2, 4};
    int result = find_the_recurring_character(array);
    
    if (result != -1) {
        std::cout << "The first recurring character is: " << result << std::endl;
    } else {
        std::cout << "No recurring character found." << std::endl;
    }

    return 0;
}

File: Hash Tables/first_recurring_char.py
def find_the_recurring_character(nums: list):
    dicio = {}
    for i in range(len(nums)):
        if nums[i] in dicio:
            return nums[i]
        dicio[nums[i]] = True

    return False

array = [2,5,1,2,3,1,2,4]
print(find_the_recurring_character(array))
